infra/lib/ci-cluster.ts

```diff
@@ -34,38 +37,59 @@ export class CICluster extends Construct {
 
     const clusterVersion = eks.KubernetesVersion.V1_23;
 
-    const mngProps: blueprints.MngClusterProviderProps = {
-      minSize: 2,
-      maxSize: 8,
-      desiredSize: 2,
-      diskSize: 150,
-      version: clusterVersion,
-      instanceTypes: [
-        ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.InstanceSize.XLARGE8),
+    const subnetTagPattern = `${STACK_NAME}/${CLUSTER_CONSTRUCT_NAME}/${CLUSTER_NAME}/${CLUSTER_NAME}-vpc/PrivateSubnet*`;
+    const securityGroupTagPattern = "kubernetes.io/cluster/" + CLUSTER_NAME;
+
+    const karpenterAddonProps = {
+      requirements: [
+          {
+            key: 'node.kubernetes.io/instance-type',
+            op: "In" as const,
+            vals: ['m5.xlarge','m5.2xlarge','m5.4xlarge','m5.8xlarge'],
+          },
+          {
+            key: 'kubernetes.io/arch',
+            op: "In" as const,
+            vals: ['amd64']
+          },
+          {
+            key: 'karpenter.sh/capacity-type',
+            op: "In" as const,
+            vals: ['on-demand']
+          },
       ],
-      amiType: eks.NodegroupAmiType.AL2_X86_64,
-      nodeGroupCapacityType: eks.CapacityType.ON_DEMAND,
-    };
+      subnetTags: { "Name": subnetTagPattern },
+      securityGroupTags: { securityGroupTagPattern : "owned" },
+      taints: [{
+        key: "workload",
+        value: "test",
+        effect: "NoSchedule" as const,
```

Comment:
I don't believe we need this taint. This would be necessary if we wanted certain workloads to *not run* on the Karpenter nodes. In our case we don't discriminate between pods

Reply:
Taint removed